pkg/sqlstore/impl/user/rowstojson.go

```diff
@@ -35,18 +35,22 @@ func getColumnsData(rows *sql.Rows) ([]sqlstore.UserColumn, error) {
 	return columns, nil
 }
 
-func getRowsData(rows *sql.Rows, numColumns int) ([][]interface{}, error) {
-	rowsData := make([][]interface{}, 0)
+func getRowsData(rows *sql.Rows, numColumns int) ([][]*sqlstore.UserValue, error) {
+	rowsData := make([][]*sqlstore.UserValue, 0)
 	for rows.Next() {
-		scanArgs := make([]interface{}, numColumns)
-		for i := range scanArgs {
-			scanArgs[i] = new(interface{})
+		vals := make([]*sqlstore.UserValue, numColumns)
+		for i := range vals {
+			val := &sqlstore.UserValue{}
+			vals[i] = val
+		}
+		scanArgs := make([]interface{}, len(vals))
+		for i := range vals {
+			scanArgs[i] = vals[i]
 		}
 		if err := rows.Scan(scanArgs...); err != nil {
 			return nil, fmt.Errorf("scan row column: %s", err)
 		}
-		rowsData = append(rowsData, scanArgs)
+		rowsData = append(rowsData, vals)
 	}
-
 	return rowsData, nil
```

Comment:
This is strongly typed now.

Reply:
Hey @jsign give this another look over in the morning. I reverted to using `UserValue` and also realized that `UserRows.Rows` can now be `[][]*UserValue` which had a nice effect bubble up and eliminate places where we were asserting types from `interface{}`.